app/controllers/api/braintree_controller.rb

```diff
@@ -13,6 +13,28 @@ def transaction
     end
   end
 
+  def webhook
+    webhook_notification = Braintree::WebhookNotification.parse(params[:bt_signature], params[:bt_payload])
+    if webhook_notification.kind == Braintree::WebhookNotification::Kind::SubscriptionChargedSuccessfully
+      card_num = webhook_notification.subscription.transactions.last.credit_card_details.last_4
+      query_params = {
+          is_subscription: true,
+          email: webhook_notification.subscription.transactions.last.customer_details.email,
+          card_num: card_num.nil? ? 'PYPL' : card_num,
+          amount: webhook_notification.subscription.transactions.last.amount.to_s
+      }
+      action = Action.where('form_data @> ?', query_params.to_json).last
```

Comment:
Yeah, basically. We need to know what page to assign the recurring donation to, so that the page in question can receive credit for generating that donation in the first place.